app/models/invoice.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class Invoice < ApplicationRecord
+  belongs_to :order, class_name: 'Spree::Order'
+  serialize :data, Hash
+  before_validation :serialize_order
+
+  def presenter
+    @presenter ||= Invoice::DataPresenter.new(self)
+  end
+
+  def serialize_order
+    return data unless data.empty?
+
+    self.data = Invoice::OrderSerializer.new(order).serializable_hash
```

Comment:
Just curious, when is the data not nil but empty?

Reply:
@mkllnk 
```ruby
3.0.3 :003 > {}.nil?
 => false 
3.0.3 :004 > {}.empty?
 => true 
```

Reply:
Oh, of course, it's a hash.